django/forms/forms.py

```diff
@@ -191,7 +191,7 @@ def add_initial_prefix(self, field_name):
 
     def _html_output(self, normal_row, error_row, row_ender, help_text_html, errors_on_separate_row):
         "Output HTML. Used by as_table(), as_ul(), as_p()."
-        top_errors = self.non_field_errors()  # Errors that should be displayed above all fields.
+        top_errors = copy.deepcopy(self.non_field_errors())  # Errors that should be displayed above all fields.
```

Comment:
Shallow copy i.e. `self.non_field_errors().copy()` should be enough. IMO.

It doesn't work properly because `ErrorList` is a subclass of `UserList` but we can fix this easily with
```python
def copy(self):
    copy = super().copy()
    copy.error_class = self.error_class
    return copy
```